rdt/hyper_transformer.py

```diff
@@ -351,7 +351,7 @@ def update_transformers_by_sdtype(
 
         for field, field_sdtype in self.field_sdtypes.items():
             if field_sdtype == sdtype:
-                self.field_transformers[field] = transformer_instance
+                self.field_transformers[field] = deepcopy(transformer_instance)
```

Comment:
This makes so we don't used the passed transformer, but since that's gonna be deprecated, I don't think we should bother writing code for it.

Reply:
what do you mean by writing code for it?

Reply:
The motivation behind this PR was that we should directly use the instance the user passes us. By deepcopying it, we are not quite doing that (so the user may be confused why the instance they passed was not affected).

However, since we plan do deprecate the scenario where a user can pass an instance directly to `update_transformers_by_sdtype`, I don't think we should bother adding extra code + test cases for this scenario.